artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -420,10 +420,20 @@ public ServerConsumer createConsumer(final long consumerID,
       }
 
       if (browseOnly) {
-         securityCheck(binding.getAddress(), CheckType.BROWSE, this);
+         try {
+            securityCheck(binding.getAddress(), CheckType.BROWSE, this);
+         }
+         catch (Exception e) {
+            securityCheck(binding.getAddress().concat(".").concat(queueName), CheckType.BROWSE, this);
```

Comment:
I agree with concatenating the address with the queueName, as long as is documented and would make sense for users.


I'm a bit confused on why you would need to catch the exception and redo the verification. I would think you would need the opposite.. in case it passed? In case the address verification failed.


I guess I don't fully understand the use case. we should talk through IRC so we can interact a bit better.